test/demo/disconnect.inttest.ts

```diff
@@ -0,0 +1,170 @@
+import 'jest';
+import {Identity} from 'openfin/_v2/main';
+
+import {AppWindow} from '../../src/provider/model/AppWindow';
+import {Application} from '../../src/client/directory';
+import {SERVICE_IDENTITY} from '../../src/client/internal';
+import {Model} from '../../src/provider/model/Model';
+import {ChannelHandler} from '../../src/provider/controller/ChannelHandler';
+import {EventHandler} from '../../src/provider/controller/EventHandler';
+import {IntentHandler} from '../../src/provider/controller/IntentHandler';
+import {FDC3ChannelEventType} from '../../src/client/main';
+
+import * as fdc3Remote from './utils/fdc3RemoteExecution';
+import {fin} from './utils/fin';
+import {OFPuppeteerBrowser} from './utils/ofPuppeteer';
+import {setupTeardown, quitApps, TestAppData} from './utils/common';
+import {testManagerIdentity, testAppInDirectory1, testAppWithPreregisteredListeners1, testAppNotInDirectory1, testAppNotFdc3} from './constants';
+import {RemoteChannel} from './utils/RemoteChannel';
+import {delay} from './utils/delay';
+
+setupTeardown();
+
+export type ProviderWindow = Window & {
+    model: Model;
+    intentHandler: IntentHandler;
+    channelHandler: ChannelHandler;
+    eventHandler: EventHandler;
+}
+
+const ofBrowser = new OFPuppeteerBrowser();
+const TEST_INTENT = 'TestIntent';
+let redChannel: RemoteChannel;
+let blueChannel: RemoteChannel;
+
+describe('Disconnecting windows', () => {
+    describe('Directory Apps', () => {
+        async function open(app: TestAppData) {
+            await fdc3Remote.open(testManagerIdentity, app.name);
+        }
+
+        testSuite('When app is closed', testAppInDirectory1, open, async (app) => {
+            await quitApps(app);
+        });
+
+        testSuite('When app is closed with preregistered listeners', testAppWithPreregisteredListeners1, open, async (app) => {
+            await quitApps(app);
+        });
+
+        testSuite('When app navigates away', testAppInDirectory1, open, async (app) => {
+            await navigateTo(app, 'about:blank');
+        });
+
+        testSuite('When app with pregistered listeners navigates away', testAppWithPreregisteredListeners1, open, async (app) => {
+            await navigateTo(app, 'about:blank');
+        });
+    });
+
+    describe('Non Directory Apps', () => {
+        async function open() {
+            await fin.Application.startFromManifest(testAppNotInDirectory1.manifestUrl);
+        }
+
+        testSuite('When app is closed', testAppNotInDirectory1, open, async (app) => {
+            await quitApps(app);
+        });
+
+        testSuite('When app navigates away', testAppNotInDirectory1, open, async (app) => {
+            await navigateTo(app, 'about:blank');
+        });
+    });
+});
+
+async function testSuite(
```

Comment:
Does `describe.each` work for this setup? Seems like the more idiomatic way of doing this kind of parameterisation.